version: '3.8'

networks:
  profiles_db_network:
    name: "profiles_db_network"

services:
  profiles_db:
    restart: unless-stopped
    image: postgres:16.0-alpine3.18
    hostname: profiles_db
    volumes:
      - ./.container_data/database/postgres/data:/var/lib/postgresql/data
      - ./db/init-up.sql:/docker-entrypoint-initdb.d/init.up.sql
    ports:
      - 5438:5432
    networks:
      - profiles_db_network
    environment :
      - PGDATA=/var/lib/postgresql/data
      - POSTGRES_USER=${POSTGRES_USER}
      - PGUSER=${PGUSER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_DB=profiles_db
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready"]
      interval: 10s
      timeout: 5s
      retries: 5

